### PR TITLE
Fix undefined check on sharingSets property of info.json

### PR DESCRIPTION
**Pull request by SethPoulsen**

### CODE REVIEW

**Comment by SethPoulsen** [comment]
By the way, does anyone know why the linter didn't catch this, possibly doing a method call on something undefined?

This is the sort of thing I assumed it was catching so I want to make sure I don't make this mistake again.

**Comment by SethPoulsen** [comment]
Thanks @jonatanschroeder . Rookie typescript mistake by me I guess. I've added it now.

File: apps/prairielearn/src/sync/sharing.ts [hunk 1]
```diff
@@ -164,7 +164,7 @@ export async function checkInvalidSharingSetRemovals(
 
     question.sharing_sets.forEach((sharingSet) => {
       // TODO: allow if the sharing set hasn't been shared to a course
-      if (!courseData.questions[question.qid].data?.sharingSets.includes(sharingSet)) {
+      if (!(courseData.questions[question.qid].data?.sharingSets ?? []).includes(sharingSet)) {
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/sync/sharing.ts:167]:
```suggestion
      if (!courseData.questions[question.qid].data?.sharingSets?.includes(sharingSet)) {
```

**Review by nwalters512 — APPROVED** [review]
FWIW, most of the types in the `course-db` file don't accurately reflect nullability/optionality. At some point, we should switch from using JSON schemas to Zod and then derive the types from the schemas as a source of truth.

**Comment by SethPoulsen** [comment]
> FWIW, most of the types in the course-db file don't accurately reflect nullability/optionality
Ya, I did realize this.

> we should switch from using JSON schemas to Zod and then derive the types from the schemas as a source of truth.

I was thinking the exact same thing as I was realizing it would have prevented this bug in the first place :)

**Pull request merged**